transport/src/main/java/io/netty/channel/PendingWriteQueue.java

```diff
@@ -166,38 +166,35 @@ public void removeAndFail(Throwable cause) {
     public ChannelFuture removeAndWriteAll() {
         assert ctx.executor().inEventLoop();
 
-        if (size == 1) {
-            // No need to use ChannelPromiseAggregator for this case.
-            return removeAndWrite();
-        }
-        PendingWrite write = head;
-        if (write == null) {
-            // empty so just return null
+        if (isEmpty()) {
```

Comment:
yes... Also the problem is that we basically need to ensure we can add more writes during the loop.